tests/test_os.py

```diff
@@ -500,34 +500,34 @@ async def test_abspath():
     assert result == abs_filename
 
 
-@pytest.mark.parametrize(("path",), [(".",), ("..",)])
-async def test_walk(path: str):
-    """Test the `walk` call."""
+def _act_on_error(oserror):
+    print(f"aiofiles.os.walk onerror handler: {oserror}")
 
-    os_walk_res = list(walk(top=path))
-    aio_walk_res = [r async for r in aiofiles.os.walk(top=path)]
 
-    assert aio_walk_res == os_walk_res
-
-
-async def test_walk_non_blocking():
-    """Test if the `walk` is non-blocking."""
-
-    async def _set_answer():
-        nonlocal val
-        nonlocal ans
-
-        val = ans
-        await asyncio.sleep(1)
-
-    async def _walk_away():
-        nonlocal val
-        nonlocal ans
-
-        async for _ in aiofiles.os.walk(top="."):
```

Comment:
`async for` is blocking by nature.

Reply:
But if these coros are swapped in the gather tool, i.e., `await asyncio.gather(_set_answer(), _walk_away())` (please refer to https://github.com/stankudrow/aiofiles/pull/1/files#diff-6e32f64be5ec084386460bf639dc465839a6b9bdc9a11b7709c497c51f5f8b36L533 ), this test passes.